components/datadog/agent/helm.go

```diff
@@ -55,25 +74,56 @@ func NewHelmInstallation(e config.CommonEnvironment, args HelmInstallationArgs,
 		return nil, err
 	}
 
+	opts = append(opts, utils.PulumiDependsOn(secret))
+
 	// Compute some values
 	agentImagePath := DockerAgentFullImagePath(&e, "")
 	agentImagePath, agentImageTag := utils.ParseImageReference(agentImagePath)
 
 	clusterAgentImagePath := DockerClusterAgentFullImagePath(&e, "")
 	clusterAgentImagePath, clusterAgentImageTag := utils.ParseImageReference(clusterAgentImagePath)
 
-	opts = append(opts, utils.PulumiDependsOn(ns, secret))
-	return helm.NewInstallation(e, helm.InstallArgs{
+	linux, err := helm.NewInstallation(e, helm.InstallArgs{
+		RepoURL:     DatadogHelmRepo,
+		ChartName:   "datadog",
+		InstallName: installName + "-linux",
+		Namespace:   args.Namespace,
+		ValuesYAML:  args.ValuesYAML,
+		Values:      buildLinuxHelmValues(installName, agentImagePath, agentImageTag, clusterAgentImagePath, clusterAgentImageTag),
+	}, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	helmComponent.LinuxHelmReleaseName = linux.Name
+	helmComponent.LinuxHelmReleaseStatus = linux.Status
+
+	windows, err := helm.NewInstallation(e, helm.InstallArgs{
```

Comment:
Should we make it conditional based on Args when there are no Windows nodes it will create objects for nothing.

Reply:
@vboulineau This change https://github.com/DataDog/test-infra-definitions/pull/126/commits/8b78a6e870013e6d3af0ba45f21425fdc3c06ead should ensure we deploy Windows agents only on clusters where there are Windows nodes.